aospy/data_loader.py

```diff
@@ -439,12 +481,20 @@ def __init__(self, template=None, data_direc=None, data_dur=None,
                 self.preprocess_func = preprocess_func
             else:
                 self.preprocess_func = template.preprocess_func
+            if upcast_float32 is not None:
```

Comment:
> Why None here but True in the other data loaders? I think True would work here also.

The distinction is that to support the construction of a `GFDLDataLoader` via a template `GFDLDataLoader` we need a neutral default value.  This allows us to unambiguously identify if the user wants to use value from the template, or override it. 

Consider the case where `upcast_float32` was `False` in the template.  If we used a default value of `True` we would not know if the user simply did not specify an `upcast_float32` argument (indicating that they would want to use the template's value) or wanted to override it (i.e. instead of `False` use `True`).